pkg/parser/custom/packagefile/packagefile.go

```diff
@@ -38,14 +44,51 @@ type PackageDetails struct {
 	CycloneDxRef *cdx.Component     `json:"cylcone_dx_ref,omitempty"`
```

Comment:
`PackageDetails` is referring to `CycloneDX` which makes me think that this data structure probably belongs within the boundary of `cyclonedxsbom` package and not in a package of its own. What do you think?

Reply:
No,  it is in the boundary of Custom parsers, and specially spdx and cyclonedx both. In my view, we should have built a wrapper over lockfile package details as we need to maintain more state than that is provided by it. PackageDetails have enough context including source of conversion so that we can convert it to more richer reports later.